core/src/main/scala/kafka/server/LogDirFailureChannel.scala

```diff
@@ -48,6 +52,7 @@ class LogDirFailureChannel(logDirNum: Int) extends Logging {
    * set of offline log dirs and enqueue it to the logDirFailureEvent queue
    */
   def maybeAddOfflineLogDir(logDir: String, msg: => String, e: IOException): Unit = {
+    numOfflineLogDirsDetected += 1;
```

Comment:
Do we have / need a code where numOfflineLogDirsDetected is cleared if they come back online?

Reply:
The expectation is once a disk failure happens, Kafka should shutdown the process. So basically Kafka regard the disk failures as "**non-recoverable**" failures (at least for non JBOD cases). It is expected that soon after this  numOfflineLogDirsDetected value increases, the server should shutdown and stop emitting this metric. So if we are seeing this metrics keeps growing or stay at non zero, something unexpected must have been happened, and we need to be alerted and manually kill the broker.